### DISCUSSION TITLE
ScriptReference/EditorGUI.DrawPreviewTexture

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#50] [Notes]
# ScriptReference/EditorGUI.DrawPreviewTexture

https://docs.unity3d.com/ScriptReference/EditorGUI.DrawPreviewTexture.html

**Comment by vertxxyz** [comment]
Drawing with this function will flicker if drawn using a PropertyDrawer. Your property drawer can override [`CanCacheInspectorGUI`](https://docs.unity3d.com/ScriptReference/PropertyDrawer.CanCacheInspectorGUI.html) and return `false`, which fixes the issue.

**Comment by vertxxyz** [comment]
If you want to draw transparent textures with this method you can use this material:
```csharp
new Material(EditorGUIUtility.LoadRequired("Previews/PreviewTransparent.shader") as Shader) {hideFlags = HideFlags.HideAndDontSave};
```
